web/src/pages/AnalyticsPage.tsx

```diff
@@ -1,7 +1,193 @@
-import React from 'react';
+import React, { useState, useEffect, useContext } from 'react';
+import { LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Label, ResponsiveContainer } from 'recharts';
+import { APIContext } from '../context/APIContext';
+
+import '../App.css';
+import styles from './AnalyticsPage.module.css';
 
 export const AnalyticsPage: React.FC = () => {
+  const [bookAnalytics, setBookAnalytics] = useState<{
+    id: string, title: string; date: string; week: number; month: number
+  }[]>([]);
+  const [clicksGraphData, setClicksGraphData] = useState<{ day: string; clicks: number; }[]>([]);
+  const [weekView, setWeekView] = useState(true);
+  const [highestView, setHighestView] = useState(true);
+
+  const client = useContext(APIContext);
+
+  // generates mm/dd/yyyy date string from timestamp string
+  function dateToString(dateStr: string): string {
+    const date = new Date(dateStr);
+    const [month, day, year] = [date.getMonth(), date.getDate(), date.getFullYear()];
+    return month + "/" + day + "/" + year;
+  }
+
+  // generates mm/dd date string for day offset from current day
+  function dateFromToday(offset: number): string {
+    const date = new Date();
+    date.setDate(date.getDate() - offset);
+    const [month, day] = [date.getMonth(), date.getDate()];
+    return month + "/" + day;
```

Comment:
Oh, this needs to be `(month + 1) + '/' + day`. Months are 0-indexed in JS (don't ask...)